source/blog/2015-05-08-rspec-compsable-matchers.md

```diff
@@ -0,0 +1,216 @@
+---
+title: How RSpec 3.0 composable matchers work
+author: Sam Phippen
+---
+
+##Warmup: the `===` protocol
+
+In Ruby, a number of objects implement the `===` protocol. Some of the objects
+that implement this protocol include:
+
+* all class objects
+* lambdas
+* regexes
+* ranges
+
+Conceptually, this protocol represents a "match". That is, the object that
```

Comment:
I've also found it useful to explain this protocol in terms of it defining a _category_ to which some objects belong and some don't -- e.g. `/foo/` defines a category to which strings containing the substring "foo" belong and all other objects do not belong.  Not sure if that's a better or worse way to describe it but it's kinda nice to not use the "match" terminology for everything.